filters/filters.go

```diff
@@ -113,12 +113,12 @@ type Filter interface {
 // the Filter instances are created using the Spec objects found in the
 // registry.
 type Spec interface {
-
-	// The name of the Spec is used to identify filters in a route definition.
+	// Name gives the name of the Spec. It is used to identify filters in a route definition.
 	Name() string
 
-	// Creates a Filter instance. Called with the parameters in the route
-	// definition while initializing a route.
+	// CreateFilter creates a Filter instance. Called with the parameters in the route
+	// definition while initializing a route. It may return a nil Filter if for a given
```

Comment:
@aryszka @szuecs 

I am here proposing the possibility for a `Spec` to return `nil` when `CreateFilter` is called, if it judges that for the current `config`, there is nothing to be performed.

My first approach was to create a custom `noopFilter` in my package, but I believe this approach (`nil`) to be a lot more performant, since it avoids calls to `Request` and `Response` on a filter that will anyways perform absolutely nothing.

Before I go forward with this, though, I would ask for your blessing.

Reply:
- the CreateFilter is not running on the hot path, so returning a nop filter is not a performance issue regarding the time, and only a minimal one regarding the space. Having a nop filter nil is also an option, methods of a nil of the right type can be called, if the methods don't reference the receiver.
- on the hot path, i don't see a big difference between checking for nil or calling a nop method
- the scope of this case is limited, typically filters are enabled/disabled by adding them or not adding them to the configuration. I see that how this is nice in this particular case, but I would rather discourage it for the typical case

so for these reasons, I would rather prefer not to do this.